app/api/services/suppliers.py

```diff
@@ -43,6 +43,23 @@ def get_supplier_by_code(self, code):
             .one_or_none()
         )
 
+    def get_supplier_by_abn(self, abn):
+        return (
+            db
+            .session
+            .query(
+                Supplier
+            )
+            .options(
+                joinedload(Supplier.domains)
+                .joinedload(SupplierDomain.domain)
```

Comment:
Do you need to load these domains?

Reply:
This was copied from `get_supplier_by_code` above - I didn't test whether it raised any n+1 issues without it, but given this may be used in similar contexts as `get_supplier_by_code` I figured it's worth maintaining the same return.